contracts/utils/SpenderLibEIP712.sol

```diff
@@ -0,0 +1,46 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.7.6;
+
+library SpenderLibEIP712 {
+    struct SpendWithPermit {
+        address tokenAddr;
+        address requester;
```

Comment:
I think the `requester` can be replaced by `msg.sender` in spender. Otherwise, it depends on each strategy contract to check whether `requester == address(this)`.

Reply:
Yes, according to 1091928080 Confluence Spec book (by @NIC619 )
My understanding is that the `requester` needs to make sure that the user is using the correct strategy contract.

Reply:
Spender needs to make sure `msg.sender == requester` so it will just replace it with `msg.sender` in `spendFromUserToWithPermit`. But for the struct here I think it is still needed for signing library and to make sure `requester` is not missing in the signing payload?

Reply:
Oh right, this should be included.

Reply:
How about the `user` field? I think strategy contract is awared of it and it can be derived from `ecrecover()` in spender.

Reply:
I think `user` is inherited from Permit's standard where `owner` is also included in the signing payload.